src/util/cmdline_definition.h

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************\
+
+Module: Command line parsing
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+// \file Command line options definition
+
+#ifndef CPROVER_UTIL_CMDLINE_DEFINITION_H
+#define CPROVER_UTIL_CMDLINE_DEFINITION_H
+
+#include <list>
+#include <string>
+#include <vector>
+
+#include "optional.h"
+
+class json_arrayt;
+class xmlt;
+
+struct cmdline_optiont
+{
+  struct argumentt
+  {
+    argumentt(std::string name, std::string type);
+
+    std::string name;
+    std::string type;
+  };
+
+  typedef std::string deprecatedt;
+
+  cmdline_optiont(
+    std::string name,
+    optionalt<argumentt> argument,
+    std::string help_text,
+    std::string option_group,
+    optionalt<deprecatedt> deprecated);
+
+  std::string name;
+  optionalt<argumentt> argument;
+  std::string help_text;
+  std::string option_group;
+  optionalt<deprecatedt> deprecated;
```

Comment:
I would like to propose one more member: `hidden`. At the moment, we have a number of command-line options that are intentionally undocumented, because either we consider them harmful (such as `--no-propagation`) or because they used to be supported and we just don't want to break scripts or other tooling still using them (such as `--floatbv`).